internal/transports/p2p/peer/peer.go

```diff
@@ -94,12 +157,26 @@ func NewPeer(
 }
 
 func (p *Peer) Connect() error {
-	conn, err := net.Dial(p.addr.Network(), p.addr.String())
+	var conn net.Conn
+	var err error
+
+	if p.inbound {
+		conn, err = p.listener.Accept()
+	} else {
+		conn, err = net.Dial(p.addr.Network(), p.addr.String())
+	}
```

Comment:
🤔 Maybe it's a good idea to leave establishing connection to some higher component. 
In other words:
server (or any other component in the future) will establish the connection either outbound connection by Dialing or inbound connection by accepting, and pass this connection in a "constructor".
This way, inbound flag could be actually only informative for now, and later used for "connection management" to choose best outbound peers.
Additionally this will enable creation of unit tests with mocked connection instead of making it at the real network level.